sdks/java/io/mongodb/src/test/java/org/apache/beam/sdk/io/mongodb/MongoDbIOTest.java

```diff
@@ -113,56 +118,27 @@ public void testFullRead() throws Exception {
         .isEqualTo(1000L);
 
-    PAssert.thatSingleton(
-        output.apply("Filter Einstein", Filter.by(new ScientistFilter("Einstein")))
-        .apply("Count Einstein", Count.<String>globally())).isEqualTo(100L);
-    PAssert.thatSingleton(
-        output.apply("Filter Darwin", Filter.by(new ScientistFilter("Darwin")))
-        .apply("Count Darwin", Count.<String>globally())).isEqualTo(100L);
-    PAssert.thatSingleton(
-        output.apply("Filter Copernicus", Filter.by(new ScientistFilter("Copernicus")))
-        .apply("Count Copernicus", Count.<String>globally())).isEqualTo(100L);
-    PAssert.thatSingleton(
-        output.apply("Filter Pasteur", Filter.by(new ScientistFilter("Pasteur")))
-        .apply("Count Pasteur", Count.<String>globally())).isEqualTo(100L);
-    PAssert.thatSingleton(
-        output.apply("Filter Curie", Filter.by(new ScientistFilter("Curie")))
-        .apply("Count Curie", Count.<String>globally())).isEqualTo(100L);
-    PAssert.thatSingleton(
-        output.apply("Filter Faraday", Filter.by(new ScientistFilter("Faraday")))
-        .apply("Count Faraday", Count.<String>globally())).isEqualTo(100L);
-    PAssert.thatSingleton(
-        output.apply("Filter Newton", Filter.by(new ScientistFilter("Newton")))
-        .apply("Count Newton", Count.<String>globally())).isEqualTo(100L);
-    PAssert.thatSingleton(
-        output.apply("Filter Bohr", Filter.by(new ScientistFilter("Bohr")))
-        .apply("Count Bohr", Count.<String>globally())).isEqualTo(100L);
-    PAssert.thatSingleton(
-        output.apply("Filter Galilei", Filter.by(new ScientistFilter("Galilei")))
-        .apply("Count Galilei", Count.<String>globally())).isEqualTo(100L);
-    PAssert.thatSingleton(
-        output.apply("Filter Maxwell", Filter.by(new ScientistFilter("Maxwell")))
-        .apply("Count Maxwell", Count.<String>globally())).isEqualTo(100L);
-
-    pipeline.run();
-  }
-
-  private static class ScientistFilter implements SerializableFunction<String, Boolean> {
-
-    private String scientist;
-
-    public ScientistFilter(String scientist) {
-      this.scientist = scientist;
-    }
-
-    public Boolean apply(String input) {
-      Document bson = Document.parse(input);
-      if (bson.getString("scientist").equals(scientist)) {
-        return true;
+    PAssert.that(output
+        .apply("Map Scientist", MapElements.via(new SimpleFunction<String, KV<String, Void>>() {
+          public KV<String, Void> apply(String input) {
+            Document bson = Document.parse(input);
+            return KV.of(bson.getString("scientist"), null);
+          }
+        }))
+        .apply("Count Scientist", Count.<String, Void>perKey())
+    ).satisfies(new SerializableFunction<Iterable<KV<String, Long>>, Void>() {
+      @Override
+      public Void apply(Iterable<KV<String, Long>> input) {
+        Iterator<KV<String, Long>> iterator = input.iterator();
```

Comment:
Use a regular Java foreach loop: for (KV<String, Long> element : input) { assertEquals(100L, element.getValue()); }